### PR TITLE
Fix bug in drks summary results and add assertion

### PR DESCRIPTION
**Pull request by maia-sh**
👋 @delwen ! Working on the trackvalue, I found a bug for the DRKS trials which have a summary results date (which we manually added), but having `has_summary_results` = FALSE. I corrected it here and added an assertion, to avoid this mishap reoccurring. I can't figure out quite how the bug got in there.

This also impacted one trackvalue report card, DRKS00011584, which we should re-run (once we pull in this corrected data) and can send out (either now or with the next reminder).

### CODE REVIEW

**Comment by delwen** [comment]
@maia-sh: thanks for this, and well spotted! I saw the new assertions but didn’t see where the actual bug was fixed. Can you point me to where that was done?

**Comment by maia-sh** [comment]
Hi @delwen !

I got back to the checks we discussed on Monday, and overall, I have no further insight into why the code didn't change, and also why crossreg rds files did, but I checked the data itself, and it looks good. Feel free to poke around some more, but otherwise I'm inclined to merge, so we can get the fix into the main dataset, and therefore into trackvalue. In any case, deferring to you for the next step (i.e., more checks, merge, do manually in trackvalue for now).

Here is the code I ran for the checks

```r
library(dplyr)

# Get main and fixed versions
trials_fix <- readr::read_rds("https://github.com/maia-sh/intovalue-data/blob/fix-drks-references/data/processed/trials.rds?raw=true")

trials_main <- readr::read_rds("https://github.com/maia-sh/intovalue-data/blob/main/data/processed/trials.rds?raw=true")


# Look at summary results date
# In main only, no result even though result date for drks
trials_fix %>%
  count(registry, has_summary_results, !is.na(summary_results_date))

trials_main %>%
  count(registry, has_summary_results, !is.na(summary_results_date))

# Look at trackvalue trial DRKS00011584
# Result date in both, but no result in main
trials_fix %>%
  filter(id == "DRKS00011584") %>%
  select(id, has_summary_results, summary_results_date)

trials_main %>%
  filter(id == "DRKS00011584") %>%
  select(id, has_summary_results, summary_results_date)

# Compare main and fixed to check for any additional changes
waldo::compare(trials_main, trials_fix)


# Unexpected data changed during fix, according to pr: registry-crossreg.rds, cross-registrations.rds, n-cross-registrations.rds
# Look into this...
# ...Can't find any differences, so not sure why appears changed in pr

# Get main and fixed versions
cr_fix <- readr::read_rds(
  # "https://github.com/maia-sh/intovalue-data/blob/fix-drks-references/data/processed/registries/registry-crossreg.rds?raw=true"
  # "https://github.com/maia-sh/intovalue-data/blob/fix-drks-references/data/processed/trn/cross-registrations.rds?raw=true"
  "https://github.com/maia-sh/intovalue-data/blob/fix-drks-references/data/processed/trn/n-cross-registrations.rds?raw=true"
)

cr_main <- readr::read_rds(
  # "https://github.com/maia-sh/intovalue-data/blob/main/data/processed/registries/registry-crossreg.rds?raw=true"
  # "https://github.com/maia-sh/intovalue-data/blob/main/data/processed/trn/cross-registrations.rds?raw=true"
  "https://github.com/maia-sh/intovalue-data/blob/main/data/processed/trn/n-cross-registrations.rds?raw=true"
)

waldo::compare(cr_main, cr_fix)
all.equal(cr_main, cr_fix)

```

**Comment by maia-sh** [comment]
ok! it's fixed. i forgot to run something from a branch we previously merged with main. I re-ran main and pushed and rebased and force pushed and tada. All good! Just added the assertions. I believe we're good to merge!

**Comment by delwen** [comment]
Hey @maia-sh, thanks for looking into this! Before I merge, what branch was it that had previously been merged with main and for which you had to run something? And what script had to be run? I'm trying to figure out if this should also have been implemented in the dashboard version.

That also brings me to this: we should think about which changes made after the dashboard release should ideally also be incorporated in the dashboard dataset, and how to do that. For example, we might want to keep the data in the current release but just incorporate some of the fixes. If that's too difficult, we might just need to do a full update of the dashboard data that contains the updates to the data (April/May) and all fixes.

**Comment by maia-sh** [comment]
> 

I'm looking at the commit history and see most recently:
6989cf42f5cbfacba5c571268f97ebfc9900de00
And 6927a7b7252b318bcb74639cc4821e93059242ea (this is a pr)

Is that what you mean?

Happy to talk through the dashboard updates question; I think it makes sense to use the most recent date in the dashboard (and we can make changes since still in pr). We may want to hold off until the iv checks are in, etc.

**Comment by delwen** [comment]
I'm still not seeing where the DRKS bug was fixed

**Review by delwen — APPROVED** [review]
We discussed this and the issue was that a script hadn't been run in main. Therefore this PR is just adding assertions.

**Pull request merged**
